components/style/traversal.rs

```diff
@@ -466,63 +463,75 @@ fn compute_style<E, D>(_traversal: &D,
 {
     context.thread_local.statistics.elements_styled += 1;
     let shared_context = context.shared;
-    // Ensure the bloom filter is up to date.
-    let dom_depth = context.thread_local.bloom_filter
-                           .insert_parents_recovering(element, traversal_data.current_dom_depth);
 
-    // Update the dom depth with the up-to-date dom depth.
-    //
-    // Note that this is always the same than the pre-existing depth, but it can
-    // change from unknown to known at this step.
-    traversal_data.current_dom_depth = Some(dom_depth);
-
-    context.thread_local.bloom_filter.assert_complete(element);
-
-    // Check to see whether we can share a style with someone.
-    let sharing_result = if element.parent_element().is_none() {
-        StyleSharingResult::CannotShare
-    } else {
-        unsafe { element.share_style_if_possible(&mut context.thread_local.style_sharing_candidate_cache,
-                                                 shared_context, &mut data) }
-    };
+    // TODO(emilio): Make cascade_input less expensive to compute in the cases
+    // we don't need to run selector matching.
```

Comment:
Yeah. I'm not super wild about the structure here, but I'm ok with fixing it up later.